lib/Compilation.js

```diff
@@ -2420,26 +2420,27 @@ Or do you want to use the entrypoints '${name}' and '${runtime}' independently o
 	 * @returns {void}
 	 */
 	removeReasonsOfDependencyBlock(module, block) {
-		const chunkGraph = this.chunkGraph;
-		const iteratorDependency = d => {
-			if (!d.module) {
-				return;
-			}
-			if (d.module.removeReason(module, d)) {
-				for (const chunk of chunkGraph.getModuleChunksIterable(d.module)) {
-					this.patchChunksAfterReasonRemoval(d.module, chunk);
-				}
-			}
-		};
-
 		if (block.blocks) {
 			for (const b of block.blocks) {
 				this.removeReasonsOfDependencyBlock(module, b);
 			}
 		}
 
 		if (block.dependencies) {
-			for (const dep of block.dependencies) iteratorDependency(dep);
+			for (const dep of block.dependencies) {
+				const originalModule = this.moduleGraph.getModule(dep);
+				if (originalModule) {
+					this.moduleGraph.removeConnection(dep);
+
+					if (this.chunkGraph) {
```

Comment:
Noticed that `chunkGraph` is only created in the `seal` phase, and that `finishModules` is called before that, hence the `if`. 

Not sure if this is entirely right however.